abn/core/service.go

```diff
@@ -37,6 +37,24 @@ var (
 
 // Start is entry point to configure services and start them
 func Start() {
+	w := initializeServer()
+	stopCh := make(chan struct{})
+
+	// start watchers
+	go w.Start(stopCh)
+
+	// launch gRPC server to respond to frontend requests
+	go launchGRPCServer([]grpc.ServerOption{})
+
+	// shutdown gracefully on termination
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGTERM, os.Interrupt)
+
+	<-sigCh
+	close(stopCh)
```

Comment:
The  `stopCh` helps to gracefully shut down the informers. 

Similarly, we might also want to add a few lines in `launchGRPCServer` func to gracefully shut down the grpc server.

[Here](https://gist.github.com/embano1/e0bf49d24f1cdd07cffad93097c04f0a) is an example ...

```go
	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", 8080))
	if err != nil {
		log.Fatalf("failed to listen: %v", err)
	}
	grpc := grpc.NewServer()
	pb.RegisterGreeterServer(grpc, &gSrv)

	sigCh := make(chan os.Signal, 1)
	signal.Notify(sigCh, os.Interrupt, syscall.SIGTERM)
	wg := sync.WaitGroup{}
	wg.Add(1)
	go func() {
		s := <-sigCh
		log.Printf("got signal %v, attempting graceful shutdown", s)
		cancel()
		grpc.GracefulStop()
		// grpc.Stop() // leads to error while receiving stream response: rpc error: code = Unavailable desc = transport is closing
		wg.Done()
	}()

	log.Println("starting grpc server")
	err = grpc.Serve(lis)
	if err != nil {
		log.Fatalf("could not serve: %v", err)
	}
	wg.Wait()
	log.Println("clean shutdown")
```

Reply:
done